src/models/StyledComponent.js

```diff
@@ -181,14 +181,17 @@ export default (ComponentStyle: Function, constructWithOptions: Function) => {
       attrs,
     } = options
 
+    const styledComponentId = (options.displayName && options.componentId) ?
+      `${options.displayName}-${options.componentId}` : componentId
```

Comment:
@philpl Not sure I follow. I believe that, because I'm using `options.displayName` and `options.componentId`, this behavior will be achieved because options isn't mutated by the default restructuring value. If I am thinking about this correctly, it's covered by the `should be generated from displayName + hash` test case.  
  
I might be missing something totally obvious. Sorry!